src/hash/rescue/rpo/mod.rs

```diff
@@ -164,6 +165,12 @@ impl Hasher for Rpo256 {
         RpoDigest::new(state[DIGEST_RANGE].try_into().unwrap())
     }
 
+    fn merge_many(values: &[Self::Digest]) -> Self::Digest {
+        let elements: Vec<Felt> =
+            Self::Digest::digests_as_elements(values.iter()).copied().collect();
+        Self::hash_elements(&elements)
```

Comment:
Can we avoid allocating vectors here? Maybe we could do something similar to what is done in winterfell? For example, add a new method to `Digest` which would return a slice that can be passed in directly to `Self::hash_elements().`

Reply:
Makes sense. I have added the same implementations that we have in Winterfell here.